src/Tools/RapidResponseDesk/sections/AreaTime/AreaAndTimeSection.jsx

```diff
@@ -1,19 +1,234 @@
+    );
+  };
+
+  useEffect(() => {
+    store.dispatch(areaAndTimeSectionSlice.actions.setTimespanArray(timespanArray));
+  }, [timespanArray]);
+
+  const getAndSetNextPrevDate = async (direction, selectedDay, id, timespanFrame, isFrom = true) => {
+    let newMoment;
+    if (direction === 'prev') {
+      newMoment = moment.utc(selectedDay).add(-1, 'days');
+    } else {
+      newMoment = moment.utc(selectedDay).add(1, 'days');
+    }
+
+    if (isFrom && (newMoment < minDateRange || newMoment > timespanFrame.to)) {
+      throw Error('invalidDateRange');
+    }
+
+    if (!isFrom && (newMoment > maxDateRange || newMoment < timespanFrame.from)) {
+      throw Error('invalidDateRange');
+    }
+
+    if (detectIfDateIsOverlapping(id, newMoment)) {
+      throw Error('invalidDateRange');
+    }
+
+    isFrom
+      ? updateTimespan(id)(newMoment, timespanFrame.to)
+      : updateTimespan(id)(timespanFrame.from, newMoment);
+  };
+
+  const setAdditionalDateRange = (currentTimespanFrame) => {
+    setTimespanArray((prevState) => {
+      return [
+        ...prevState,
+        {
+          from: currentTimespanFrame.from.clone().add(-1, 'days').startOf('day'),
+          to: currentTimespanFrame.from.clone().add(-1, 'days').endOf('day'),
+          id: timespanArray.length,
+          displayCalendarFrom: false,
+          displayCalendarTo: false,
+        },
+      ];
+    });
+  };
+
+  const updateTimespan = (id) => (fromTime, toTime) => {
+    setTimespanArray((prevState) => {
+      return prevState.map((item, index) => (index === id ? { ...item, from: fromTime, to: toTime } : item));
+    });
+  };
+
+  const closeAllCalendarDialogs = () => {
+    setTimespanArray((prevState) => {
+      return prevState.map((item, index) => ({
+        ...item,
+        displayCalendarFrom: false,
+        displayCalendarTo: false,
+      }));
+    });
+  };
+
+  const updateCalendarOpenState = (selectedTimespan, state, isFrom = true) => {
+    if (state) {
+      closeAllCalendarDialogs();
+    }
+
+    setTimespanArray((prevState) => {
+      return prevState.map((item, index) =>
+        index === selectedTimespan.id
+          ? {
+              ...item,
+              displayCalendarFrom: isFrom ? state : false,
+              displayCalendarTo: !isFrom ? state : false,
+            }
+          : item,
+      );
+    });
+  };
+
+  const removeTimespanFromList = (id) => {
+    setTimespanArray((prevState) => prevState.filter((item, index) => index !== id));
+  };
+
+  const setTimespanPicker = (timespan, index) => (
+    <TimespanPicker
+      id="aoi-time-select"
+      minDate={minDateRange}
+      maxDate={maxDateRange}
+      datePickerInputStyle={{ width: '85px' }}
+      timespan={{ fromTime: timespan.from, toTime: timespan.to }}
+      applyTimespan={updateTimespan(index)}
+      timespanExpanded={true}
+      calendarHolder={cardHolderRef}
+      timespanLimit={(selectedTimespan) => detectIfDateIsOverlapping(index, selectedTimespan)}
```

Comment:
What about toast messages? Could this be useful for this project?

Reply:
We do kind of use some sort of toast messages (see image below), you can test it by going to browser.dataspace.copernicus.eu and check it out on the top, but for all other errors/warnings we use the "error panel" on the left side, so I would stick to using this. Also discussed this with Daniel and his opinion is the same.

![image](https://github.com/user-attachments/assets/d33311aa-9eb2-437a-a0cb-02a7cc2f9434)

Reply:
If it is allowed to select a date which overlaps another time span, does it mean that i need to create a flag, which blocks search button and is released when correction is made?

Reply:
That might be the best yes.

I have 2 reasons for all of these changes:
1. if we allow overlapping timespans and since we don't know exactly yet how backend will handle this, it might happen that results could get duplicated
2. for me personally it is annoying that if I click on a date that would overlap, simply nothing happens. I would much rather have behaviour that I select some timespan and I get notified that there is an error there - a simple warning saying something like "timespan's should not overlap" should suffice and with what you suggested it is even better. (then keep in mind that you might want to set a title on the button explaining why the button is disabled)

Reply:
Make sense yeah. Good points.
What I've done now is to check nonstop if there is an overlap. If it is, it shows a message at the top of order panel, if it isn't it hides it from the top.

![image](https://github.com/user-attachments/assets/97393626-2c28-4ad7-a3cf-93c658051ed8)

Reply:
As for the warning message, i would suggest something like: "Selected time ranges overlap". What do you think?

Reply:
Yeah, that sounds good and also looks good 👍

Reply:
It looks good, but I still have a small request 😅 

Could you move the orange "X" button a bit more to the left, it feels out of line.

Also please make the notification in the message-panel a bit higher (min-height: 30px; should do the trick)

Reply:
I will do this yeah. However I implemented an canBeClosed flag which hides x button on demand now. Maybe in this case this message is always visible as long as there is a range overlap?

Reply:
Another comment from Daniel: like you suggested somewhere above (I can't seem to find it..), the "Search" button should be disabled in case of overlapping timeranges.

The warning text should be changed to: "Please select non-overlapping time ranges."

Reply:
> Maybe in this case this message is always visible as long as there is a range overlap

This makes sense yes

Reply:
Should be possible to show more than one message in errorPanel?

Reply:
I don't see a use case where two errors would happen at the same time, so for now I would say to just leave it as it is to only support 1 error message at time.

Reply:
There is still a bug with overlapping timespans if one timespan is "inside" the other timespan.

How to reproduce:
1. first timespan 01.07.2024 - 30.07.2024
2. second timespan 30.06.2024 - 31.07.2024

This should produce an error, but it does not